cpp/include/typedb/user/UserManager.hpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include "typedb/common/native.hpp"
+
+namespace TypeDB {
+
+class UserManager {
+    private:
+     TypeDBNative::UserManager* userManagerNative;
+
+    public:
+     UserManager(const TypeDBNative::Connection*);
+     UserManager(const UserManager&) = delete;
+     ~UserManager();
```

Comment:
Rule of five violation: https://en.cppreference.com/w/cpp/language/rule_of_three
```suggestion
     UserManager(const UserManager&) = delete;
     UserManager& operator=(const UserManager&) = delete;
     UserManager(UserManager&&) = default;
     UserManager& operator=(UserManager&&) = default;
     ~UserManager();
```